File: 841-keys-and-rooms/841-keys-and-rooms.cpp
class Solution {
public:
    void dfs(int curr , vector<int> &vis , vector<vector<int>> &rooms){
        vis[curr] = 1;
        for(auto x : rooms[curr]){
            if(!vis[x]) dfs(x,vis,rooms);
        }
    }
    bool canVisitAllRooms(vector<vector<int>>& rooms) {
        int n = rooms.size();
        vector<int> vis(n,0);
        dfs(0,vis,rooms);
        for(int i = 0 ; i < n;  i++) if(!vis[i]) return false;
        return true;
    }
};

File: 841-keys-and-rooms/841-keys-and-rooms.py
class Solution:
    def dfs(self,node,rooms,vis):
        vis[node]=True
        for i in rooms[node]:
            if not vis[i]:
                vis[i]=True
                self.dfs(i,rooms,vis)
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        vis=[False]*len(rooms)
        self.dfs(0,rooms,vis)
        if False in vis:
            return False
        return True
        